test/parallel/proofOfStake.3rdPartyRewards.activation.check.test.ts

```diff
@@ -0,0 +1,378 @@
+          liqId,
+          Assets.DEFAULT_AMOUNT,
+          newToken3,
+        ),
+        testUser.keyRingPair,
+      ).then((events) => {
+        const res = getEventResultFromMangataTx(events);
+        expect(res.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+      });
+
+      await waitForRewards(testUser, liqId, 20, newToken3);
+
+      testUser.addAsset(newToken);
+      testUser.addAsset(newToken3);
+      await testUser.refreshAmounts(AssetWallet.BEFORE);
+      const rewardToken1Before =
+        await testUser.getAsset(newToken)?.amountBefore.free!;
+      const rewardToken3Before =
+        await testUser.getAsset(newToken3)?.amountBefore.free!;
+
+      await signTx(
+        getApi(),
+        await ProofOfStake.claim3rdpartyRewards(liqId, newToken),
+        testUser.keyRingPair,
+      );
+
+      await signTx(
+        getApi(),
+        await ProofOfStake.claim3rdpartyRewards(liqId, newToken3),
+        testUser.keyRingPair,
+      );
+
+      await testUser.refreshAmounts(AssetWallet.AFTER);
+      const rewardToken1After =
+        await testUser.getAsset(newToken)?.amountAfter.free!;
+      const rewardToken3After =
+        await testUser.getAsset(newToken3)?.amountAfter.free!;
+
+      expect(rewardToken1Before).bnEqual(BN_ZERO);
+      expect(rewardToken3Before).bnEqual(BN_ZERO);
+      expect(rewardToken1After).bnGt(BN_ZERO);
+      expect(rewardToken3After).bnGt(BN_ZERO);
+    });
+
+    test("GIVEN promoted pool MGX-Token2, pool MGX-Token1 AND user with activated 3rd party rewards of pool MGX-Token2 for Token1 WHEN user deactivates 3rd party rewadrs with too many liquidity tokens THEN receive error", async () => {
+      await createAndPromoteSecondPool(Assets.DEFAULT_AMOUNT.muln(2), true);
+      await signTx(
+        getApi(),
+        await ProofOfStake.activateLiquidityFor3rdpartyRewards(
+          liqId,
+          Assets.DEFAULT_AMOUNT,
+          newToken,
+        ),
+        testUser.keyRingPair,
+      ).then((events) => {
+        const res = getEventResultFromMangataTx(events);
+        expect(res.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+      });
+
+      await waitForRewards(testUser, liqId, 20, newToken);
+
+      await signTx(
+        getApi(),
+        await ProofOfStake.deactivateLiquidityFor3rdpartyRewards(
+          liqId,
+          Assets.DEFAULT_AMOUNT.muln(2),
+          newToken,
+        ),
+        testUser.keyRingPair,
+      ).then((events) => {
+        const res = getEventResultFromMangataTx(events);
+        expect(res.state).toEqual(ExtrinsicResult.ExtrinsicFailed);
+        expect(res.data).toEqual("NotEnoughAssets");
+      });
+    });
+  });
+});
+
+async function createAndPromoteSecondPool(
+  userLiqTokenAmount: BN,
+  isRewardPoolEnabled: boolean,
+) {
+  await Sudo.batchAsSudoFinalized(
+    Assets.mintToken(newToken, sudo, Assets.DEFAULT_AMOUNT.muln(40e6)),
+    Assets.mintToken(liqId, testUser, userLiqTokenAmount),
```

Comment:
plz do not use global variables in a local scope. send them as a parameter in the function.

Reply:
added var in function